packages/extensions/venia-sample-payments-checkmo/talons/useBillingAddress.js

```diff
@@ -0,0 +1,316 @@
+import { useQuery, useApolloClient, useMutation } from '@apollo/client';
+import { useCartContext } from '@magento/peregrine/lib/context/cart';
+
+/**
+ * Maps address response data from GET_BILLING_ADDRESS and GET_SHIPPING_ADDRESS
+ * queries to input names in the billing address form.
+ * {@link creditCard.gql.js}.
+ *
+ * @param {ShippingCartAddress|BillingCartAddress} rawAddressData query data
+ */
+export const mapAddressData = rawAddressData => {
+    if (rawAddressData) {
+        const {
+            firstName,
+            lastName,
+            city,
+            postcode,
+            phoneNumber,
+            street,
+            country,
+            region
+        } = rawAddressData;
+
+        return {
+            firstName,
+            lastName,
+            city,
+            postcode,
+            phoneNumber,
+            street1: street[0],
+            street2: street[1],
+            country: country.code,
+            region: region.code
+        };
+    } else {
+        return {};
+    }
+};
+
+/**
+ * Talon to handle Credit Card payment method.
+ *
+ * @param {Boolean} props.shouldSubmit boolean value which represents if a payment nonce request has been submitted
+ * @param {Function} props.onBillingAddressChangedError callback to invoke when error was throw for set the billing address
+ * @param {Function} props.onBillingAddressChangedSuccess callback to invoke when address was sucessfully set
+ * @param {DocumentNode} props.queries.getBillingAddressQuery query to fetch billing address from cache
+ * @param {DocumentNode} props.queries.getIsBillingAddressSameQuery query to fetch is billing address same checkbox value from cache
+ * @param {DocumentNode} props.mutations.setBillingAddressMutation mutation to update billing address on the cart
+ *
+ * @returns {
+ *   errors: Map<String, Error>,
+ *   isBillingAddressSame: Boolean,
+ *   initialValues: {
+ *      firstName: String,
+ *      lastName: String,
+ *      city: String,
+ *      postcode: String,
+ *      phoneNumber: String,
+ *      street1: String,
+ *      street2: String,
+ *      country: String,
+ *      state: String,
+ *      isBillingAddressSame: Boolean
+ *   },
+ *   shippingAddressCountry: String,
+ * }
+ */
+export const useBillingAddress = (props = {}) => {
+    const {
+        queries,
+        mutations,
+        shouldSubmit,
+        onBillingAddressChangedError,
+        onBillingAddressChangedSuccess
+    } = props;
+
+    const {
+        getBillingAddressQuery,
+        getIsBillingAddressSameQuery,
+        getShippingAddressQuery
+    } = queries;
+
+    const { setBillingAddressMutation } = mutations;
+
+    const [errors, setErrors] = useState([]);
+
+    const client = useApolloClient();
+    const formState = useFormState();
+    const { validate: validateBillingAddressForm } = useFormApi();
+    const [{ cartId }] = useCartContext();
+
+    const { data: billingAddressData } = useQuery(getBillingAddressQuery, {
```

Comment:
I would suggest changing this to a lazy query instead and call the billing address query only if `isBillingAddressSame` is false. In most cases, this will save on the network.

Reply:
I dont knwo how i solve this is same in `useCreditCard`